modules/elastic/src/main/scala/ESClient.scala

```diff
@@ -50,40 +81,79 @@ object ESClient:
     private def unitOrFail[A](response: Response[A]): F[Unit] =
       response.fold(MonadThrow[F].raiseError[Unit](response.error.asException))(_ => MonadThrow[F].unit)
 
+    val indexAttributeKey = AttributeKey.string("index")
+    val sizeAttributeKey  = AttributeKey.long("size")
+
     def search[A](query: A, from: From, size: Size)(using q: Queryable[A]): F[List[Id]] =
-      client
-        .execute(q.searchDef(query)(from, size))
-        .flatMap(toResult)
-        .map(_.hits.hits.toList.map(h => Id(h.id)))
+      searchDuration
+        .recordDuration(TimeUnit.MILLISECONDS, Attribute(indexAttributeKey, q.index(query).value))
+        .surround:
+          client
+            .execute(q.searchDef(query)(from, size))
+            .flatMap(toResult)
+            .map(_.hits.hits.toList.map(h => Id(h.id)))
+            .onError(_ => searchErrorCounter.inc(Attribute(indexAttributeKey, q.index(query).value)))
 
     def count[A](query: A)(using q: Queryable[A]): F[Long] =
-      client
-        .execute(q.countDef(query))
-        .flatMap(toResult)
-        .map(_.count)
+      countDuration
+        .recordDuration(TimeUnit.MILLISECONDS, Attribute(indexAttributeKey, q.index(query).value))
+        .surround:
+          client
+            .execute(q.countDef(query))
+            .flatMap(toResult)
+            .map(_.count)
+            .onError(_ => countErrorCounter.inc(Attribute(indexAttributeKey, q.index(query).value)))
```

Comment:
Everything below doesn't necessarily apply to your service and infrastructure. But it might be useful in the future.

___ 

You can use attributes to distinguish errored/succeeded actions. The OpenTelemetry [Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/database/database-metrics/) encourages this approach.

For example, `count.duration` indicates how long it takes to execute the `count` query. If you add `error.type` attribute, you can track successful and unsuccessful queries within the same metric.

In Grafana, you can query data as:
```
count_duration_count{"error_type" != ""} # shows the number of failed queries
count_duration_count{"error_type" = ""} # shows the number of succeeded queries
```

Code:
```scala
def withErrorType(static: Attributes)(ec: Resource.ExitCase) = ec match
  case Resource.ExitCase.Succeeded =>
    static
  case Resource.ExitCase.Errored(e) =>
    static.added(Attribute("error.type", e.getClass.getName))
  case Resource.ExitCase.Canceled =>
    static.added(Attribute("error.type", "canceled"))

countDuration
  .recordDuration(
    TimeUnit.MILLISECONDS, 
    withErrorType(Attributes(Attribute(indexAttributeKey, q.index(query).value)))
  )
  .surround:
    client
      .execute(q.countDef(query))
      .flatMap(toResult)
      .map(_.count)
```

-----

If we take it one step further, we can follow the OTel specification:
```scala
opDuration <- meter.histogram[Double]("db.client.operation.duration").withUnit("s").create

def search[A](query: A, from: From, size: Size)(using q: Queryable[A]): F[List[Id]] =
  opDuration
    .recordDuration(
      TimeUnit.SECONDS, 
      withErrorType(Attributes(Attribute("db.operation.name", "search"), Attribute(indexAttributeKey, q.index(query).value)))
    )
    .surround:
      ...

def count[A](query: A)(using q: Queryable[A]): F[Long] =
  opDuration
    .recordDuration(
      TimeUnit.SECONDS, 
      withErrorType(Attributes(Attribute("db.operation.name", "count"), Attribute(indexAttributeKey, q.index(query).value)))
    )
    .surround:
      ...

```

And the queries:
```
db_client_operation_duration_count{"db_operation_name" = "search",  "error_type" == ""}
db_client_operation_duration_count{"db_operation_name" = "search",  "error_type" != ""}
db_client_operation_duration_count{"db_operation_name" = "count",  "error_type" == ""}
db_client_operation_duration_count{"db_operation_name" = "count",  "error_type" != ""}
```

Reply:
oh this is extremely useful, I think I could apply it right now. Thanks a lot @iRevive.

Also do you mind bump `otel4s-experimental-metric` version to match otel4s `0.11-8e1f500-SNAPSHOT`? This is binary incompatible with the previous one somehow.

Reply:
Here is a new build: `0.4.0-6-8c1230f-SNAPSHOT`.